packages/devtools_app/lib/src/shared/table/column_widths.dart

```diff
@@ -121,17 +121,22 @@ extension TreeColumnWidthExtension<T extends TreeNode<T>>
         //
         // `characterWidth` was determined through trial and error to roughly
         // approximate the width of a single character.
-        const characterWidth = 9.0;
+        final characterWidth = scaleByFontFactor(9.0);
         double longestWidth = 0;
         for (var node in flattenedList) {
           final width = column.getNodeIndentPx(node) +
-              (column.getDisplayValue(node).length *
-                  scaleByFontFactor(characterWidth));
+              (column.getDisplayValue(node).length * characterWidth);
           if (width > longestWidth) {
             longestWidth = width;
           }
         }
-        width = longestWidth;
+        // Add two characters to account for expansion caret and ensure we
+        // always display the full column title.
+        width = max(
+              max(longestWidth, column.minWidthPx ?? 0),
+              column.title.length * characterWidth,
+            ) +
+            characterWidth * 2;
```

Comment:
do we need to estimate here or do we know the exact size of the expansion caret?

Reply:
This is the exact size (one character for `^` and one for the whitespace).